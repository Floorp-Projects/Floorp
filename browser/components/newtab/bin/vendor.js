#!/usr/bin/env node
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

/* eslint-disable no-console */

const { cp, set } = require("shelljs");
const path = require("path");

const filesToVendor = {
  // XXX currently these two licenses are identical.  Perhaps we should check
  // in case that changes at some point in the future.
  "react/LICENSE": "REACT_AND_REACT_DOM_LICENSE",
  "react/umd/react.production.min.js": "react.js",
  "react/umd/react.development.js": "react-dev.js",
  "react-dom/umd/react-dom.production.min.js": "react-dom.js",
  "react-dom/umd/react-dom.development.js": "react-dom-dev.js",
  "react-dom/umd/react-dom-server.browser.production.min.js":
    "react-dom-server.js",
  "react-redux/LICENSE.md": "REACT_REDUX_LICENSE",
  "react-redux/dist/react-redux.min.js": "react-redux.js",
  "react-transition-group/dist/react-transition-group.min.js":
    "react-transition-group.js",
  "react-transition-group/LICENSE": "REACT_TRANSITION_GROUP_LICENSE",
};

set("-v"); // Echo all the copy commands so the user can see what's going on
for (let srcPath of Object.keys(filesToVendor)) {
  cp(
    path.join("node_modules", srcPath),
    path.join("vendor", filesToVendor[srcPath])
  );
}

console.log(`
Check to see if any license files have changed, and, if so, be sure to update
https://searchfox.org/mozilla-central/source/toolkit/content/license.html`);
