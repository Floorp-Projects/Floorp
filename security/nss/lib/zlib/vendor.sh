#!/usr/bin/env bash
# Vendor a new version of zlib.
#
# Note: This script doesn't remove files if they are removed in the zlib release.

set -e

if [[ $# -lt 1 ]]; then
    echo "Usage: $0 <version>" 1>&2
    exit 2
fi

version="$1"
ztmp="zlib.$version"

cd "$(dirname "$0")"
../../fuzz/config/git-copy.sh https://github.com/madler/zlib "v$version" "$ztmp"
fullversion="$version ("$(cat "$ztmp"/.git-copy)")"
sed -i -e 's/^Version: .*/Version: '"$fullversion"'/' README.nss

prune=(
    .git-copy
    .gitignore
    CMakeLists.txt
    ChangeLog
    FAQ
    INDEX
    Makefile
    Makefile.in
    amiga
    configure
    contrib
    doc
    example.c
    examples
    make_vms.com
    minigzip.c
    msdos
    nintendods
    old
    os400
    qnx
    test
    treebuild.xml
    watcom
    win32
    zconf.h.cmakein
    zconf.h.in
    zlib.3
    zlib.3.pdf
    zlib.map
    zlib.pc.cmakein
    zlib.pc.in
    zlib2ansi
)
for i in "${prune[@]}"; do rm -rf "$ztmp"/"$i"; done
for i in "$ztmp"/*; do mv "$i" .; done
rmdir "$ztmp"
