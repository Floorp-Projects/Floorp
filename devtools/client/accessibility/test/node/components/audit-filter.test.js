/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
"use strict";

const { mount } = require("enzyme");

const { createFactory } = require("devtools/client/shared/vendor/react");
const { span } = require("devtools/client/shared/vendor/react-dom-factories");
const Provider = createFactory(
  require("devtools/client/shared/vendor/react-redux").Provider
);

const ConnectedAuditFilterClass = require("devtools/client/accessibility/components/AuditFilter");
const AuditFilterClass = ConnectedAuditFilterClass.WrappedComponent;
const AuditFilter = createFactory(ConnectedAuditFilterClass);
const {
  setupStore,
} = require("devtools/client/accessibility/test/node/helpers");
const { FILTERS } = require("devtools/client/accessibility/constants");

const {
  accessibility: { SCORES },
} = require("devtools/shared/constants");

describe("AuditController component:", () => {
  it("audit filter not filtered", () => {
    const store = setupStore();

    const wrapper = mount(Provider({ store }, AuditFilter({}, span())));
    expect(wrapper.html()).toMatchSnapshot();

    const filter = wrapper.find(AuditFilterClass);
    expect(filter.children().length).toBe(1);
    expect(filter.childAt(0).is("span")).toBe(true);
  });

  it("audit filter filtered no checks", () => {
    const store = setupStore({
      preloadedState: { audit: { filters: { [FILTERS.CONTRAST]: true } } },
    });

    const wrapper = mount(Provider({ store }, AuditFilter({}, span())));
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it("audit filter filtered unknown checks", () => {
    const store = setupStore({
      preloadedState: { audit: { filters: { tbd: true } } },
    });

    const wrapper = mount(Provider({ store }, AuditFilter({}, span())));
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it("audit filter filtered contrast checks success", () => {
    const store = setupStore({
      preloadedState: { audit: { filters: { [FILTERS.CONTRAST]: true } } },
    });

    const wrapper = mount(
      Provider(
        { store },
        AuditFilter(
          {
            checks: {
              CONTRAST: {
                value: 5.11,
                color: [255, 0, 0, 1],
                backgroundColor: [255, 255, 255, 1],
                isLargeText: false,
                score: SCORES.AA,
              },
            },
          },
          span()
        )
      )
    );
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.isEmptyRender()).toBe(true);
  });

  it("audit filter filtered contrast checks fail", () => {
    const store = setupStore({
      preloadedState: { audit: { filters: { [FILTERS.CONTRAST]: true } } },
    });

    const CONTRAST = {
      value: 3.1,
      color: [255, 0, 0, 1],
      backgroundColor: [255, 255, 255, 1],
      isLargeText: false,
      score: SCORES.FAIL,
    };

    const wrapper = mount(
      Provider(
        { store },
        AuditFilter(
          {
            checks: { CONTRAST },
          },
          span()
        )
      )
    );
    expect(wrapper.html()).toMatchSnapshot();
    const filter = wrapper.find(AuditFilterClass);
    expect(filter.children().length).toBe(1);
    expect(filter.childAt(0).is("span")).toBe(true);
  });

  it("audit filter filtered contrast checks fail range", () => {
    const store = setupStore({
      preloadedState: { audit: { filters: { [FILTERS.CONTRAST]: true } } },
    });

    const CONTRAST = {
      min: 1.19,
      max: 1.39,
      color: [128, 128, 128, 1],
      backgroundColorMin: [219, 106, 116, 1],
      backgroundColorMax: [156, 145, 211, 1],
      isLargeText: false,
      score: SCORES.FAIL,
    };

    const wrapper = mount(
      Provider(
        { store },
        AuditFilter(
          {
            checks: { CONTRAST },
          },
          span()
        )
      )
    );
    expect(wrapper.html()).toMatchSnapshot();
    const filter = wrapper.find(AuditFilterClass);
    expect(filter.children().length).toBe(1);
    expect(filter.childAt(0).is("span")).toBe(true);
  });
});
