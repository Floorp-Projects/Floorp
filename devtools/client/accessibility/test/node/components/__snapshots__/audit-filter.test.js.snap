// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditController component: audit filter filtered contrast checks fail 1`] = `"<span></span>"`;

exports[`AuditController component: audit filter filtered contrast checks fail range 1`] = `"<span></span>"`;

exports[`AuditController component: audit filter filtered contrast checks success 1`] = `""`;

exports[`AuditController component: audit filter filtered no checks 1`] = `""`;

exports[`AuditController component: audit filter filtered unknown checks 1`] = `""`;

exports[`AuditController component: audit filter not filtered 1`] = `"<span></span>"`;
